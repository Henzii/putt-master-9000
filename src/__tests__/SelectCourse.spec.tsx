import { render } from '@testing-library/react-native';
import React from 'react';
import { Provider } from 'react-native-paper';
import SelectCourses from '../components/SelectCourse';
import mockedCourses from './mocks/mockedCourses';

const TestComp = () => <Provider><SelectCourses /></Provider>;

jest.mock('react-redux', () => {
    return {
        useDispatch: () => null
    };
});
jest.mock('../hooks/useCourses', () => {
    return () => {
        return {
            courses: mockedCourses,
            loading: false,
            addLayout: () => null,
            addCourse: () => null,
            fetchMore: () => null,
            error: null,
            rest: null
        };
    };
});

describe('<SelectCourse /> testit', () => {
    it('RenderÃ¶ityy oikein...', async () => {
        const puu = render(<TestComp />).toJSON();
        expect(puu).toMatchSnapshot();
    });
});
