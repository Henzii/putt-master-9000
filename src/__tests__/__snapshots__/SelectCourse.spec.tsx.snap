// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectCourse /> testit Renderöityy oikein... 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        undefined,
        undefined,
        undefined,
        Object {
          "flex": 1,
        },
        Object {
          "backgroundColor": "#f6f6f6",
          "padding": 0,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "padding": 10,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 0,
            "minWidth": 64,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
          testID="AddCourseButton"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "marginLeft": 12,
                  "marginRight": -4,
                }
              }
            >
              <Text />
            </View>
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Add Course
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 4,
            "borderStyle": "solid",
            "borderWidth": 0,
            "elevation": 0,
            "minWidth": 64,
          }
        }
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": undefined,
            }
          }
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 4,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Object {
                  "marginLeft": 12,
                  "marginRight": -4,
                }
              }
            >
              <Text />
            </View>
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "textAlign": "left",
                  },
                  Array [
                    Object {
                      "letterSpacing": 1,
                      "marginHorizontal": 16,
                      "marginVertical": 9,
                      "textAlign": "center",
                    },
                    undefined,
                    Object {
                      "textTransform": "uppercase",
                    },
                    Object {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    Object {
                      "fontFamily": "System",
                      "fontWeight": "500",
                    },
                    undefined,
                  ],
                ]
              }
            >
              Search
            </Text>
          </View>
        </View>
      </View>
    </View>
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      ListFooterComponent={
        <Text
          style={
            Object {
              "color": "rgba(0,0,0,0.2)",
            }
          }
        >
              No more... No más...
        </Text>
      }
      data={
        Array [
          Object {
            "distance": Object {
              "meters": 69.69,
              "string": "69 meters",
            },
            "id": "courseid1",
            "layouts": Array [
              Object {
                "holes": 9,
                "id": "layoutid1",
                "name": "TestLayout1",
                "par": 27,
                "pars": Array [
                  3,
                  3,
                  3,
                  3,
                  3,
                  3,
                  3,
                  3,
                  3,
                ],
              },
              Object {
                "holes": 9,
                "id": "layoutid2",
                "name": "TestLayout2",
                "par": 29,
                "pars": Array [
                  3,
                  4,
                  3,
                  3,
                  3,
                  3,
                  3,
                  4,
                  3,
                ],
              },
            ],
            "location": Object {
              "coordinates": Array [
                64.02,
                22.22,
              ],
            },
            "name": "TestCourse1",
          },
          Object {
            "distance": Object {
              "meters": 123,
              "string": "123 meters",
            },
            "id": "courseid2",
            "layouts": Array [
              Object {
                "holes": 18,
                "id": "layoutid3",
                "name": "TestLayout3",
                "par": 56,
                "pars": Array [
                  3,
                  3,
                  3,
                  3,
                  3,
                  3,
                  4,
                  4,
                  3,
                  3,
                  3,
                  3,
                  3,
                  3,
                  3,
                  3,
                  3,
                  3,
                ],
              },
            ],
            "location": Object {
              "coordinates": Array [
                63.02,
                21.22,
              ],
            },
            "name": "TestCourse2",
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onEndReachedThreshold={0.1}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "padding": 5,
                },
                Object {
                  "backgroundColor": "#ffffff",
                },
              ]
            }
          >
            <View>
              <View
                style={
                  Object {
                    "backgroundColor": "#f6f6f6",
                  }
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "expanded": false,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "padding": 8,
                        },
                        Array [
                          Object {
                            "padding": 5,
                          },
                          Object {
                            "backgroundColor": "#ffffff",
                          },
                        ],
                      ],
                    ]
                  }
                  testID="SingleCourse"
                >
                  <View
                    pointerEvents="none"
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "margin": 8,
                          },
                          Object {
                            "flex": 1,
                            "justifyContent": "center",
                          },
                        ]
                      }
                    >
                      <Text
                        numberOfLines={1}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            Object {
                              "textAlign": "left",
                            },
                            Array [
                              Object {
                                "fontSize": 16,
                              },
                              Object {
                                "color": "rgba(0, 0, 0, 0.87)",
                              },
                              Array [
                                Object {
                                  "fontSize": 19,
                                },
                                false,
                                false,
                              ],
                            ],
                          ]
                        }
                      >
                        TestCourse1
                      </Text>
                      <Text
                        numberOfLines={3}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            Object {
                              "textAlign": "left",
                            },
                            Array [
                              Object {
                                "fontSize": 14,
                              },
                              Object {
                                "color": "rgba(0, 0, 0, 0.54)",
                              },
                              Array [
                                Array [
                                  Object {
                                    "fontSize": 19,
                                  },
                                  false,
                                  false,
                                ],
                                Object {
                                  "fontSize": 14,
                                },
                              ],
                            ],
                          ]
                        }
                      >
                        2 layouts
                      </Text>
                    </View>
                    <View
                      style={
                        Array [
                          Object {
                            "margin": 8,
                          },
                          Object {
                            "alignItems": "center",
                            "height": 40,
                            "justifyContent": "center",
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "display": "flex",
                            "flexDirection": "row",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "display": "flex",
                              "flexDirection": "column",
                            }
                          }
                        >
                          <View
                            pointerEvents="box-none"
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "height": 40,
                                  "justifyContent": "center",
                                  "margin": 8,
                                  "width": 40,
                                },
                                Object {
                                  "color": "gray",
                                  "margin": 0,
                                  "padding": 0,
                                },
                              ]
                            }
                          >
                            <Text />
                          </View>
                          <Text
                            style={
                              Object {
                                "color": "gray",
                                "margin": 0,
                                "padding": 0,
                              }
                            }
                          >
                            69 meters
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,0.2)",
                "height": 1,
              }
            }
          />
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "padding": 5,
                },
                Object {
                  "backgroundColor": "#ffffff",
                },
              ]
            }
          >
            <View>
              <View
                style={
                  Object {
                    "backgroundColor": "#f6f6f6",
                  }
                }
              >
                <View
                  accessibilityRole="button"
                  accessibilityState={
                    Object {
                      "expanded": false,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "padding": 8,
                        },
                        Array [
                          Object {
                            "padding": 5,
                          },
                          Object {
                            "backgroundColor": "#ffffff",
                          },
                        ],
                      ],
                    ]
                  }
                  testID="SingleCourse"
                >
                  <View
                    pointerEvents="none"
                    style={
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "margin": 8,
                          },
                          Object {
                            "flex": 1,
                            "justifyContent": "center",
                          },
                        ]
                      }
                    >
                      <Text
                        numberOfLines={1}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            Object {
                              "textAlign": "left",
                            },
                            Array [
                              Object {
                                "fontSize": 16,
                              },
                              Object {
                                "color": "rgba(0, 0, 0, 0.87)",
                              },
                              Array [
                                Object {
                                  "fontSize": 19,
                                },
                                false,
                                false,
                              ],
                            ],
                          ]
                        }
                      >
                        TestCourse2
                      </Text>
                      <Text
                        numberOfLines={3}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            Object {
                              "textAlign": "left",
                            },
                            Array [
                              Object {
                                "fontSize": 14,
                              },
                              Object {
                                "color": "rgba(0, 0, 0, 0.54)",
                              },
                              Array [
                                Array [
                                  Object {
                                    "fontSize": 19,
                                  },
                                  false,
                                  false,
                                ],
                                Object {
                                  "fontSize": 14,
                                },
                              ],
                            ],
                          ]
                        }
                      >
                        1 layout
                      </Text>
                    </View>
                    <View
                      style={
                        Array [
                          Object {
                            "margin": 8,
                          },
                          Object {
                            "alignItems": "center",
                            "height": 40,
                            "justifyContent": "center",
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "display": "flex",
                            "flexDirection": "row",
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "display": "flex",
                              "flexDirection": "column",
                            }
                          }
                        >
                          <View
                            pointerEvents="box-none"
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "height": 40,
                                  "justifyContent": "center",
                                  "margin": 8,
                                  "width": 40,
                                },
                                Object {
                                  "color": "gray",
                                  "margin": 0,
                                  "padding": 0,
                                },
                              ]
                            }
                          >
                            <Text />
                          </View>
                          <Text
                            style={
                              Object {
                                "color": "gray",
                                "margin": 0,
                                "padding": 0,
                              }
                            }
                          >
                            123 meters
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
        >
          <Text
            style={
              Object {
                "color": "rgba(0,0,0,0.2)",
              }
            }
          >
                No more... No más...
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
